fiducial_vlam/CMakeLists.txt

```diff
@@ -55,27 +55,29 @@ include_directories(
 
 # Debugging: set _dump_all_variables to true
 set(_dump_all_variables false)
-if(_dump_all_variables)
+if (_dump_all_variables)
   get_cmake_property(_variable_names VARIABLES)
   list(SORT _variable_names)
-  foreach(_variable_name ${_variable_names})
+  foreach (_variable_name ${_variable_names})
     message(STATUS "${_variable_name}=${${_variable_name}}")
-  endforeach()
-endif()
+  endforeach ()
+endif ()
 
 #=============
 # vloc node
 #=============
 
-add_executable(vloc_node
+add_executable(vloc_node_main
```

Comment:
Looks like you're not building library targets. I came out of our discussion thinking that we would have these targets:
add_library(vloc_node ...) # Builds shared library
add_executable(vloc_main ...) # Links shared library

Based on my previous comment, it might also be interesting to have a 3rd target for debugging:
add_executable(vloc_debug ...) # Builds executable w/o a shared library

Thoughts?

Reply:
Right. I hadn't figured out how to do build a library.